caffe2/onnx/backend.cc

```diff
@@ -890,17 +890,47 @@ Caffe2Ops Caffe2Backend::ConvertNode(
   return OnnxNodeToCaffe2Ops(init_model, pred_model, &onnx_node, opset_version);
 }
 
+void Caffe2Backend::CheckOpSchemaArguments(
+    const caffe2::OpSchema* schema,
+    const Caffe2Ops* op) {
+  const auto schema_args = schema -> args();
+  std::vector<std::string> argnames;
+  std::transform(
+      schema_args.begin(),
+      schema_args.end(),
+      std::back_inserter(argnames),
+      [](caffe2::OpSchema::Argument elem) { return elem.name(); });
+
+  for (const auto& result_op : op -> ops) {
+    for (const auto& arg : result_op.arg()) {
+      if (std::find(argnames.begin(), argnames.end(), arg.name()) ==
+          argnames.end()) {
+        CAFFE_THROW(
+            "Don't know how to map unexpected argument ",
+            arg.name(),
+            " (from operation ",
+            result_op.name(), ")");
+      }
+    }
+  }
+}
+
 Caffe2Ops Caffe2Backend::OnnxNodeToCaffe2Ops(
     const ModelProto& init_model,
     const ModelProto& pred_model,
     OnnxNode* onnx_node,
     int opset_version) {
+  Caffe2Ops res;
   if (get_special_operators().count(onnx_node->node.op_type())) {
-    return (this->*get_special_operators().at(onnx_node->node.op_type()))(
+    res = (this->*get_special_operators().at(onnx_node->node.op_type()))(
         onnx_node, opset_version);
   } else {
-    return CommonOnnxNodeToCaffe2Ops(onnx_node, opset_version);
+    res = CommonOnnxNodeToCaffe2Ops(onnx_node, opset_version);
   }
+  const auto op_type = onnx_node -> node.op_type();
+  const auto* schema = OpSchemaRegistry::Schema(op_type);
```

Comment:
here should lookup the schema of the c2 op, not the onnx node (their name can be different).